sympy/printing/ccode.py

```diff
@@ -219,8 +219,9 @@ def _print_ITE(self, expr):
         return self._print(_piecewise)
 
     def _print_MatrixElement(self, expr):
-        return "{0}[{1}]".format(expr.parent, expr.j +
-                expr.i*expr.parent.shape[1])
+        from sympy.printing.precedence import PRECEDENCE
+        return "{0}[{1}]".format(self.parenthesize(expr.parent,PRECEDENCE["Add"]),
```

Comment:
Sorry, I was wrong about the precedence, it's not `Add`, it's `Atom`.

Anyway, I'm not sure about the C code. Did you check if the programs coming out actually do compile?

You cannot add arrays in C.